test/index.test.js

```diff
@@ -132,5 +134,35 @@ describe('Compat - index.js', function () {
       });
     });
   });
-});
+  describe('hoistable functions', function () {
+    it('should work with defined hoistable function', function (done) {
+      function modify(obj) {
+        obj.anotherKey = 'what';
+        return obj;
+      }
+
+      const wmodel = new Dynastar({ model, hashKey, rangeKey, modify });
+      assume(wmodel.modify).is.a('function');
+      wmodel.findAll()
+        .on('data', function (data) {
+          const modified = wmodel.modify(data);
+          assume(modified.anotherKey).equals('what');
+        })
+        .on('error', done)
+        .on('end', done);
+    });
 
+    it('should hoist function with AwaitWrap and make them awaitable', async function () {
+      function somethingAsync(data, done) {
+        data.somethingAsync = true;
+        return void setImmediate(done, null, data);
+      }
+      const wmodel = new AwaitWrap(new Dynastar({ model, hashKey, rangeKey, somethingAsync }));
```

Comment:
actually is this what we want. I think I'd like to see a realistic example of a model here.  wouldn't you have something like:

```js
class MyModel {
  // ...other datastar fns
  somethingAsync(data, done) {
    // whatever
  }
  someOtherFunction(data) {
    // not async
  }
}
```

So to do this you'd need to do something like:

```js
const model = new MyModel();
const wmodel = new AwaitWrap(new Dynastar({ model, hashKey, rangeKey, somethingAsync: model.somethingAsync }));
```

which feels weird. should `Dynastar` just hoist all the model functions? or perhaps it should allow you to pass the hoistable directly:

```js
const model = new MyModel();
const wmodel = new AwaitWrap(new Dynastar({ model, hashKey, rangeKey, hoist: ['somethingAsync'] }));
```

Reply:
oh right, models is basically just the `joi`/datastar object here.

Reply:
the model is whats returned from `dynamo.define`, we arent defining any functions on that like we were on datastar models. This is replacing how we defined functions on the prototype of datastar models and just making them functions called given an object etc.